zebrad/src/application.rs

```diff
@@ -174,6 +188,23 @@ impl Application for ZebradApp {
 
         Ok(())
     }
+
+    fn shutdown(&mut self, shutdown: Shutdown) -> ! {
+        if let Err(e) = self.state().components.shutdown(self, shutdown) {
+            fatal_error(self, &e)
+        }
+
+        // Swap out a fake app so we can trigger the destructor on the original
+        let mut fake_app = Self::default();
+        std::mem::swap(self, &mut fake_app);
+        drop(fake_app);
```

Comment:
Suggestion: this variable has a confusing name

When we call drop, we're actually dropping the real app.

```suggestion
        // Swap out a fake app so we can trigger the destructor on the original
        let mut app = Self::default();
        std::mem::swap(self, &mut app);
        drop(app);
```